ui/app/helpers/utils/i18n-helper.test.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import assert from 'assert'
+
+describe('i18n helper', function () {
+  const TEST_LOCALE_CODE = 'TEST_LOCALE_CODE'
+
+  const TEST_KEY_1 = 'TEST_KEY_1'
+  const TEST_KEY_2 = 'TEST_KEY_2'
+  const TEST_KEY_3 = 'TEST_KEY_3'
+  const TEST_KEY_4 = 'TEST_KEY_4'
+  const TEST_KEY_5 = 'TEST_KEY_5'
+  const TEST_KEY_6 = 'TEST_KEY_6'
+  const TEST_KEY_6_HELPER = 'TEST_KEY_6_HELPER'
+  const TEST_KEY_7 = 'TEST_KEY_7'
+  const TEST_KEY_7_HELPER_1 = 'TEST_KEY_7_HELPER_1'
+  const TEST_KEY_7_HELPER_2 = 'TEST_KEY_7_HELPER_2'
+  const TEST_KEY_8 = 'TEST_KEY_8'
+  const TEST_KEY_8_HELPER_1 = 'TEST_KEY_8_HELPER_1'
+  const TEST_KEY_8_HELPER_2 = 'TEST_KEY_8_HELPER_2'
+
+  const TEST_SUBSTITUTION_1 = 'TEST_SUBSTITUTION_1'
+  const TEST_SUBSTITUTION_2 = 'TEST_SUBSTITUTION_2'
+  const TEST_SUBSTITUTION_3 = 'TEST_SUBSTITUTION_3'
+  const TEST_SUBSTITUTION_4 = 'TEST_SUBSTITUTION_4'
+  const TEST_SUBSTITUTION_5 = 'TEST_SUBSTITUTION_5'
+
+  const testLocaleMessages = {
+    [TEST_KEY_1]: {
+      message: 'This is a simple message.',
+      expectedResult: 'This is a simple message.',
+    },
+    [TEST_KEY_2]: {
+      message: 'This is a message with a single non-react substitution $1.',
+    },
+    [TEST_KEY_3]: {
+      message: 'This is a message with a two non-react substitution $1 and $2.',
+    },
+    [TEST_KEY_4]: {
+      message: '$1 - $2 - $3 - $4 - $5',
+    },
+    [TEST_KEY_5]: {
+      message: '$1 - $2 - $3',
+    },
+    [TEST_KEY_6]: {
+      'message': 'Testing a react substitution $1.',
+    },
+    [TEST_KEY_6_HELPER]: {
+      'message': TEST_SUBSTITUTION_1,
+    },
+    [TEST_KEY_7]: {
+      'message': 'Testing a react substitution $1 and another $2.',
+    },
+    [TEST_KEY_7_HELPER_1]: {
+      'message': TEST_SUBSTITUTION_1,
+    },
+    [TEST_KEY_7_HELPER_2]: {
+      'message': TEST_SUBSTITUTION_2,
+    },
+    [TEST_KEY_8]: {
+      'message': 'Testing a mix $1 of react substitutions $2 and string substitutions $3 + $4.',
+    },
+    [TEST_KEY_8_HELPER_1]: {
+      'message': TEST_SUBSTITUTION_3,
+    },
+    [TEST_KEY_8_HELPER_2]: {
+      'message': TEST_SUBSTITUTION_4,
+    },
+  }
+  const t = getMessage.bind(null, TEST_LOCALE_CODE, testLocaleMessages)
+
+  const TEST_SUBSTITUTION_6 = (<div style={{ color: 'red' }} key="test-react-substitutions-1"> { t(TEST_KEY_6_HELPER) } </div>)
+  const TEST_SUBSTITUTION_7_1 = (<div style={{ color: 'red' }} key="test-react-substitutions-1"> { t(TEST_KEY_7_HELPER_1) } </div>)
+  const TEST_SUBSTITUTION_7_2 = (<div style={{ color: 'blue' }} key="test-react-substitutions-1"> { t(TEST_KEY_7_HELPER_2) } </div>)
+  const TEST_SUBSTITUTION_8_1 = (<div style={{ color: 'orange' }} key="test-react-substitutions-1"> { t(TEST_KEY_8_HELPER_1) } </div>)
+  const TEST_SUBSTITUTION_8_2 = (<div style={{ color: 'pink' }} key="test-react-substitutions-1"> { t(TEST_KEY_8_HELPER_2) } </div>)
+
+
+  const testConfigs = {
```

Comment:
Setting all of the test inputs and expectations here in one block makes these tests a bit harder to read, as I need to scroll back and forth through the file to check what everything means. I've used this style in the past when using the same set of inputs => outputs in multiple tests, but here there seems to be a roughly 1:1 correspondence between key/value and test, so I don't see any advantage to outweigh the cost in readability.

I'd be curious to hear if you disagree and do find this to be more readable, or if there's some other advantage I'm missing.